packages/core/src/Ignition.ts

```diff
@@ -37,15 +39,15 @@ export class Ignition {
   private _journal: ICommandJournal;
 
   constructor({
```

Comment:
A better solution would be:

1) Keep receiving services here.
2) Make the constructor `protected`.
3) Add a `static` factory that receives providers.
4) In the tests extend `Ignition` (e.g. `MockableIgnition`) with a public `constructor` or another factory.

Reply:
This saves us from fiddling with private fields later

Reply:
I have added a static factory method and made the Ignition constructor protected.

The tests use a `TestIgnition` with its own Services orientated constructor.